### PR TITLE
feat(dora): add incident related tables

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
1. [x] Add new tables: `incidents` and `incident_assignees`    
    - Records in `incidents` are from `issues`(with type = 'INCIDNET') or `pull_requests`(whose lables satisfy some rules, rules are still WIP)
2. [x] Rename table `project_issue_metrics` to `project_incident_deployment_relationships`    
3. [x] DORA plugin: add new tasks
    -  [x] `ConvertIssuesToIncidents` generates incidents from issues
    -  [ ]  `ConvertPullRequestToIncidents` generates incidents from pull_requests
4. [x] Update webhook, create/close incident directly.

### Does this close any open issues?
Closes N/A

### Screenshots
Include any relevant screenshots here.

#### incident_assignees
image

#### incidents
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Comment by d4x1** [comment]
I am not sure whether this pr should be cherry picked to release-v1.0 or not.

File: backend/core/models/domainlayer/code/pull_request.go [hunk 1]
```diff
@@ -58,3 +60,54 @@ type PullRequest struct {
 func (PullRequest) TableName() string {
 	return "pull_requests"
 }
+
+func (pr PullRequest) ConvertStatusToIncidentStatus() string {
+	switch pr.Status {
+	case OPEN:
+		return ticket.TODO
+	case CLOSED:
+		return ticket.OTHER
```
[the remark this answers is not in the document]
Comment (abeizn) [reply backend/core/models/domainlayer/code/pull_request.go:69]: And after your PR is merged, devtable will probably need to change the sql uniformly.

[the remark this answers is not in the document]
Comment (d4x1) [reply backend/core/models/domainlayer/code/pull_request.go:69]: Yes. `CLOSED` pull requests should be `OTHER`.

**Review by abeizn — APPROVED** [review]
lgtm

**Pull request merged**